src/cron/updateCurrentStreak.ts

```diff
@@ -0,0 +1,40 @@
+import { Cron } from './cron';
+import { User, UserStreak } from '../entity';
+import { checkAndClearUserStreak } from '../common';
+
+const cron: Cron = {
+  name: 'update-current-streak',
+  handler: async (con, logger) => {
+    try {
+      await con.transaction(async (entityManager): Promise<void> => {
+        const usersPastStreakTime = await entityManager
```

Comment:
Should we consider streaming the results?

Reply:
Maybe at the current scale it's fine not to. So your call

Reply:
I actually thought about it, but noticed we never really do for crons only bin scripts.
Happy to change it though.

Reply:
Let's add a metric maybe instead of many users are being returned from this query and then also how many we reset from. Since we run it every 30mins the batch should be relatively small I believe which is great

Reply:
That sounds good, it should be minimal past the first I think

Reply:
Added a counter let me know if it works. (CC: @omBratteng )
And rewrote the cron logic :)